samples/swift/uidemo/ChatViewController.swift

```diff
@@ -0,0 +1,251 @@
+//
+//  Copyright (c) 2016 Google Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import UIKit
+import Firebase
+import FirebaseDatabaseUI
+import FirebaseAuthUI
+
+struct Chat {
+  var uid: String
+  var name: String
+  var text: String
+  
+  var dictionary: [String: String] {
+    return [
+      "uid" : self.uid,
+      "name": self.name,
+      "text": self.text,
+    ]
+  }
+  
+  init(uid: String, name: String, text: String) {
+    self.name = name; self.uid = uid; self.text = text
+  }
+  
+  init?(snapshot: FIRDataSnapshot) {
+    guard let dict = snapshot.value as? [String: String] else { return nil }
+    guard let name = dict["name"] else { return nil }
+    guard let uid  = dict["uid"]  else { return nil }
+    guard let text = dict["text"] else { return nil }
+    
+    self.name = name
+    self.uid = uid
+    self.text = text
+  }
+}
+
+// View controller demonstrating using a FirebaseCollectionViewDataSource
+// to populate a collection view with chat messages. The relevant code
+// is in the call to `collectionViewDataSource.populateCellWithBlock`.
+//
+// All of the error handling in this controller is done with `fatalError`;
```

Comment:
Hah! People definitely will - can you move/replicate this comment to close to sections users are likely to cut and paste, or when errors will appear.